File: src/test/java/no/utgdev/diy/ex6/Ex6Test.java
package no.utgdev.diy.ex6;

import no.utgdev.diy.DIY;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class Ex6Test {
    @BeforeEach
    void reset() {
        DIY.reset();
    }

    @Test
    void run() {
        DIY.load("no.utgdev.diy.ex6");
        MinComponent controller = (MinComponent) DIY.getBean("MinComponent");

        assertThat(controller).isNotNull();
        assertThat(controller.hei()).isEqualTo("Hei fra MinComponent og MinService");
    }
}

File: src/test/kotlin/no/utgdev/diy/ex6/Ex6Test.kt
package no.utgdev.diy.ex6

import no.utgdev.diy.DIY
import no.utgdev.diy.DIY.Companion.getBean
import no.utgdev.diy.DIY.Companion.load
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class Ex6Test {
    @BeforeEach
    fun reset() {
        DIY.reset()
    }

    @Test
    fun run() {
        load("no.utgdev.diy.ex6")
        val controller = getBean("MinComponent") as MinComponent?
        Assertions.assertThat(controller).isNotNull
        Assertions.assertThat(controller!!.hei()).isEqualTo("Hei fra MinComponent og MinService")
    }
}